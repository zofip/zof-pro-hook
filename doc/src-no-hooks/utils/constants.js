export const APICLIENTS = 'http://ec2-54-237-229-204.compute-1.amazonaws.com:8383/json-server/clients';
export const APIPROJECTS = 'http://ec2-54-237-229-204.compute-1.amazonaws.com:8383/json-server/projects';
export const APICOMPONENTS = 'http://ec2-54-237-229-204.compute-1.amazonaws.com:8383/json-server/components';