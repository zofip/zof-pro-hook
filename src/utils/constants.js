export const API_CLIENTS = '';
export const API_PROJECTS = '';
export const API_COMPONENTS = '';
export const API_COMPONENT_BY_ID= '';
export const API_SAVE_INDICATOR = '';
export const EMPTY = '';
export const USER = 'admin';
export const PASS = '';
export const PERCENT = 'PORCENTAJE';
export const REMOVE = 'REMOVE';
export const ADD = 'ADD';
export const INFORMATION = 'INFORMATION';
export const WARNING = 'WARNING';
export const MESSAGE_INDICATORS_SAVED = 'Indicators successfully saved!!';
export const MESSAGE_NO_INDICATORS_TO_SAVE = 'No indicators to save!!'
export const TOKEN = 'token';
export const CLIENT = 'Client';
export const PROJECT = 'Project';