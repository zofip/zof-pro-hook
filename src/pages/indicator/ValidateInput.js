import React from 'react';

import NumberInput from '../../common/NumberInput';
import PorcentSlider from '../../common/PorcentSlider';
import { PERCENT } from '../../utils/constants'


const ValidateInput = ({unit, handleInputValueChange}) => {
    if(unit === PERCENT) {
        return <PorcentSlider handleInputValueChange ={handleInputValueChange} />;
    }
    return <NumberInput handleInputValueChange = {handleInputValueChange}/>;
}

export default ValidateInput;