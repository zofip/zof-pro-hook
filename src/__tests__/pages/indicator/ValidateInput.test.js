import React from 'react';
import { shallow } from 'enzyme';

import ValidateInput from '../../../pages/indicator/ValidateInput';
import { PERCENT } from '../../../utils/constants';

describe('<ValidateInput />', () => {
   it('Rendering', () => {
      const validateInputPercent = shallow(<ValidateInput unit={PERCENT}/>);
      expect(validateInputPercent).toMatchSnapshot();
      const validateInputNumber = shallow(<ValidateInput />);
      expect(validateInputNumber).toMatchSnapshot();
   });
});
