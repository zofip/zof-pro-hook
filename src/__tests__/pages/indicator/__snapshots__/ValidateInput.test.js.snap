// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ValidateInput /> Rendering 1`] = `<PorcentSlider />`;

exports[`<ValidateInput /> Rendering 2`] = `<NumberInput />`;
